syncstorage-db-common/src/util.rs

```diff
@@ -112,6 +112,29 @@ impl SyncTimestamp {
     pub fn as_rfc3339(self) -> Result<String, SyncstorageDbError> {
         to_rfc3339(self.as_i64())
     }
+
+    /// Convert this SyncTimestamp into a chrono::NaiveDateTime (UTC) for use with Diesel's `Timestamp`.
+    pub fn as_naive_datetime(self) -> Result<NaiveDateTime, SyncstorageDbError> {
+        let millis = self.as_i64();
+        let secs = millis / 1000;
+        let nsecs: u32 = ((millis % 1000) * 1_000_000).try_into().map_err(|e| {
+            SyncstorageDbError::internal(format!(
+                "Invalid timestamp (nanoseconds) {}: {}",
+                millis, e
+            ))
+        })?;
+
+        let ts = Utc.timestamp_opt(secs, nsecs);
+
+        if let Some(dt) = ts.single() {
+            Ok(dt.naive_utc())
+        } else {
+            Err(SyncstorageDbError::internal(format!(
+                "Invalid or ambiguous timestamp {}: {:?}",
+                millis, ts
+            )))
+        }
+    }
```

Comment:
I think `from_timestamp_millis` is still simpler, unless there's something I'm just not understanding here.

Reply:
I haven't used `ToSql` but it's basically the inverse of `FromSql` which we use below. I'm not 100% sure it would work as an argument to bind (e.g. `.bind<SyncTimestamp>()`) but in theory it should work for that as well. There aren't a lot of great examples of its usage other than [use within a struct](https://github.com/diesel-rs/diesel/issues/1783).

 Feel free to *not* use it, just throwing it out there as a potential option.